### PR TITLE
Fix handling of 'count' select in ddb query/scan all

### PR DESCRIPTION
**Pull request by arthurnovak**
`q_all`/`scan_all` should be able to return count value in case of setting `{select, count}` in `Opts`. At the moment it fails because there is no specific case clause for that.
AWS always returns `undefined` items for `count` request, so we'll match on that response.

@motobob @danilagamma

### CODE REVIEW

**Comment by motobob** [comment]
apparently this never worked so please add a unit test for this.

File: src/erlcloud_ddb_util.erl [hunk 1]
```diff
@@ -309,22 +311,39 @@ q_all(Table, KeyConditionsOrExpression, Opts) ->
...
+                     items              = undefined,
+                     count              = Count}} ->
+            {ok, lists:sum([Count|Acc])};
+        {ok, #ddb2_q{last_evaluated_key = undefined,
+                     items              = Items}} ->
+            {ok, flatreverse([Items|Acc])};
+        {ok, #ddb2_q{last_evaluated_key = LastKey,
+                     items              = Items,
+                     count              = Count}} ->
+            UpdAcc = case Items of
```
Comment (motobob) [inline src/erlcloud_ddb_util.erl]: seems unnecessary nesting to me. 
make it outer case clauses please by `items = undefined`

Comment (arthurnovak) [inline src/erlcloud_ddb_util.erl]: Actually it looks better for me having one case clause for case when `last_evaluated_key /= undefined`. I think it makes sense to call `q_all(Tbl, KeyConditionsOrExpression, Opts, Config, UpdAcc, LastKey)` just once in one clause.

**Comment by nalundgaard** [comment]
@arthurnovak It looks like this `applies to `scan_all`](http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Scan.html#DDB-Scan-request-Select) as well. Any reason not to also fix that?

**Comment by arthurnovak** [comment]
@nalundgaard I need just `q_all` 😄. But yeah, it makes sense to fix that behavior in `scan_all` as well. Will do

File: src/erlcloud_ddb_util.erl (hunk 1 above)
Comment (arthurnovak) [inline src/erlcloud_ddb_util.erl]: ok

**Pull request merged**